### PR TITLE
Add english translation for Locale and User Interface Language strings

### PR DESCRIPTION
**Pull request by joshua-stone**
Please do not forget to update the mmex.pot file and write information about the fixed bug in the prerelease [page](https://github.com/moneymanagerex/moneymanagerex/releases).

Addresses https://github.com/moneymanagerex/moneymanagerex/issues/5213

- - -
This change is [Reviewable](https://reviewable.io/reviews/moneymanagerex/moneymanagerex/6267)

### CODE REVIEW

**Comment by guanlisheng** [comment]
@ovari what's your opinion here ?

File: po/en_US.po [hunk 1]
```diff
@@ -6571,6 +6571,18 @@ msgstr ""
 msgid "Running on:"
 msgstr ""
 
+#: util.cpp:1495
```
Comment (vomikan) [inline po/en_US.po:6574]: There is no need to change this file. It changes automatically. You need to add new phrases only to the mmex.pot file.

**Comment by joshua-stone** [comment]
Thanks for the reviews. The PR has been updated.

File: src/util.cpp [hunk 2]
```diff
@@ -1502,7 +1502,7 @@ const wxString getProgramDescription(int type)
         if (display->IsPrimary())
         {
 
-            description << wxString::Format(bull + "%ix%i %ibit %ix%ippi\n",
+            description << wxString::Format(bull + "%ix%i %i-bit %ix%ippi\n",
```
Comment (ovari) [inline src/util.cpp:1505]: Can the newline please be added outside the translation?

From:
`description << wxString::Format(bull + "%ix%i %i-bit %ix%ippi\n",`
to:
`description << wxString::Format(bull + "%ix%i %i-bit %ix%ippi" + "\n",`

Thank you

File: po/mmex.pot [hunk 3]
```diff
@@ -54,6 +54,18 @@ msgstr ""
...
+#: util.cpp:1495
+msgid "System Locale: %s"
+msgstr ""
+
+#: util.cpp:1496
+msgid "User Interface Language: %s"
+msgstr ""
+
+#: util.cpp:1505
+msgid "%ix%i %i-bit %ix%ippi\n"
```
Comment (ovari) [inline po/mmex.pot:66]: Remove newline?

From:
`msgid "%ix%i %i-bit %ix%ippi\n"`
to:
`msgid "%ix%i %i-bit %ix%ippi"`

**Comment by joshua-stone** [comment]
More changes pushed.

**Comment by ovari** [comment]
Can the contents in the other tabs in the About dialog please be translatable?

image

Thank you

**Comment by joshua-stone** [comment]
> Can the contents in the other tabs in the About dialog please be translatable?
> 
> image
> 
> Thank you

This is getting out of scope from the original MR which pertains to the addition of Locale/Language info strings. I think it would be more appropriate to file a separate issue that's dedicated to adding translations for all tabs.

**Comment by joshua-stone** [comment]
In the meantime, I think I'd like to work on addressing a UI regression once this MR has been merged:

https://github.com/moneymanagerex/moneymanagerex/pull/6265#issuecomment-1785981429

**Comment by ovari** [comment]
@whalley can you please add the strings to [Crowdin](https://crowdin.com/project/moneymanagerex)? Thank you

**Comment by ovari** [comment]
@vomikan do you know when these strings will be added to [Crowdin](https://crowdin.com/project/moneymanagerex)? Thank you

**Comment by ovari** [comment]
@joshua-stone should the string be changed from
[%ix%i %i-bit %ix%ippi](https://github.com/moneymanagerex/moneymanagerex/pull/6267/files#diff-d724d5a0bd2ee9450a167a257401b31cd5ce2c38d3e1e3c1636210cb148f6d42R66)
to
`%1$ix%2$i %3$i-bit %4$ix%5$ippi`

Thank you

**Comment by vomikan** [comment]
it should be 
```
            description << wxString::Format(bull + _("%1$ix%2$i %3$i-bit %4$ix%5$ippi") + "\n",
```

**Comment by whalley** [comment]
Think we have discussed this before but we really should avoid having parameters in strings subject to translation as this can lead to runtime crashes if the translations are not done correctly.

**Comment by n-stein** [comment]
> Think we have discussed this before but we really should avoid having parameters in strings subject to translation as this can lead to runtime crashes if the translations are not done correctly. 

Agreed. Isn't `bit` the only potentially translated word here?

**Comment by vomikan** [comment]
> Isn't `bit` the only potentially translated word here?

The string "%1$ix%2$i %3$i-bit %4$ix%5$ippi" conteins an order of strings(numbers) for insert. I meean 1$ 2$ 3$ ...
In which locale will you need to change the word order?
• 1920x1080 32-bit 96x96ppi

**Comment by ovari** [comment]
The hyphen is also translatable so `-bit` would be the translatable part. In magyar (Hungarian), the hyphen is a (non-breaking) space.

**Comment by ovari** [comment]
Perhaps the code could be changed from

`description << wxString::Format(bull + _("%1$ix%2$i %3$i-bit %4$ix%5$ippi") + "\n",`

to

`description << wxString::Format(bull + "%1$ix%2$i " + _(%3$i-bit) + " " + _("%4$ix%5$ippi")) + "\n",`

**Comment by vomikan** [comment]
If you need to translate the words "bit" and "ppi", then you can do it differently:
https://github.com/moneymanagerex/moneymanagerex/pull/6555/commits/1a123e32f153a8e176d504627cb22b831ccdb49f

**Pull request merged**